src/librustc_mir/interpret/eval_context.rs

```diff
@@ -546,7 +546,7 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
 
         info!("ENTERING({}) {}", self.cur_frame(), self.frame().instance);
 
-        if self.stack.len() > self.tcx.sess.const_eval_stack_frame_limit {
+        if self.stack.len() > *self.tcx.sess.recursion_limit.get() {
```

Comment:
This is kind of a breaking change now. If users were using a low recursion limit (below 100), their code will now stop compiling if they were using a const eval stack deeper than their recursion limit. I think it would be best to use a minimum of 100 for the check here.

Reply:
Alternatively we could crater this... this is the kind of breaking change we can do if the diagnostics are good enough.

What is the default recursion limit?

Reply:
the default is 128, so no worries there. It would only be a breaking change if ppl explicitly set a lower value

Reply:
I'd rather not introduce a hack in the Miri engine if the number of affected crates is most likely to be 0 (and moreover, if crates actually are affected, the fix of raising the recursion limit to 100 is trivial).

Reply:
seems fine to me, too, I just wanted to bring it up. Crater seems like a waste for this. We can always add the hack later if it ends up being important

Reply:
> if the diagnostics are good enough.

Currently, I've not changed the diagnostics at all, so it doesn't mention that the limit can be changed via `#![recursion_limit]` at all. I'm not sure exactly how to make that adjustment, though.